test/base/kernel_launch_generic.cpp

```diff
@@ -468,6 +543,53 @@ void run2d_row_reduction(std::shared_ptr<gko::EXEC_TYPE> exec)
 TEST_F(KernelLaunch, ReductionRow2D) { run2d_row_reduction(exec); }
 
 
+void run2d_row_reduction_cached(std::shared_ptr<gko::EXEC_TYPE> exec,
+                                std::vector<gko::dim<2>> dims)
+{
+    // This assumes at most 256 OpenMP Threads
+    constexpr int64_t max_tmp_elems = 4 * 256;
```

Comment:
That's not a future-proof assumption, and conflicts with other executors with more parallelism. Why does the other approach (tmp storage smaller than input) not work here?

Reply:
Because for small input vectors, the temporary storage can exceed the actual vector size with the current implementations.
I can set it as a minimum and require that `tmp_storage` is either smaller than `max_tmp_elems` or smaller than input size.

Reply:
That sounds like a mistake to me. I don't remember the implementation details clearly, would it be possible to fix this easily?

Reply:
I just pushed an update that should address this for all reductions (in this test file).
I tried to change as little of the implementation as possible while limiting the temporary storage to the input size.